tests/sls/homu/valid_builders.py

```diff
@@ -0,0 +1,37 @@
+import json
+import os.path
+import subprocess
+import toml
+
+from tests.util import Failure, Success
+
+
+def run():
+    homu_cfg = toml.load('/home/servo/homu/cfg.toml')
+    homu_builders = homu_cfg['repo']['servo']['buildbot']
+
+    # We need to invoke a new process to read the buildbot master config
+    # because buildbot is written in python2.
+    scriptpath = os.path.join(os.path.dirname(__file__), 'get_buildbot_cfg.py')
+    ret = subprocess.run(
+        ['/usr/bin/python2', scriptpath],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE
+    )
+    if ret.returncode != 0:
+        return Failure(
+            'Unable to retrieve buildbot builders name:', ret.stderr
+        )
+
+    buildbot_builders = json.loads(ret.stdout.decode('utf-8'))['builders']
+
+    for buildertype in ['builders', 'try_builders']:
+        diff = set(homu_builders[buildertype]) - set(buildbot_builders)
+        if diff:
+            return Failure(
+                "Homu {} config isn't synced with buildbot config"
+                .format(buildertype),
+                'Difference: {}'.format(diff)
```

Comment:
Before the for loop, initialize a string var (e.g. `failure_msg`) to an empty string. In the loop, if the `if diff:` condition is true, append information about the diff to `failure_msg`.

After the loop, `if failure_msg` then return a `Failure` passing `failure_msg` for the second argument. This way if there invalid builders for `builders` and `try_builders` they're both reported.

Prose changes:
  - Keep the same title (first arg) for the Failure, but don't mention the builder set
   - Use a more descriptive prefix than `Difference`, maybe `Invalid builders for {builder_set}:` (this gets added to `failure_msg` iteratively)